src/test/java/e2e/pages/EmailPage.java

```diff
@@ -0,0 +1,102 @@
+package e2e.pages;
+
+import enums.SortDirections;
+import enums.SortValues;
+import org.openqa.selenium.By;
+import org.openqa.selenium.StaleElementReferenceException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+import java.util.NoSuchElementException;
+
+public class EmailPage extends ContactInfoPage {
+
+    public EmailPage(WebDriver driver) {
+        super(driver);
+    }
+
+    @FindBy(xpath = "//*[@class='nav-link active']")
+    WebElement emailTab;
+
+    @FindBy(xpath = "//*[@id='btn-add-phone']")
+    WebElement addEmailButton;
+
+    @FindBy(xpath = "//*[@id='input-email']")
+    WebElement emailDialog;
+    @FindBy(xpath = "//*[@class='btn btn-primary']")
+    WebElement saveButton;
+
+    @FindBy(xpath = "//*[@class='dropdown-item btn-email-edit']")
+    WebElement editButton;
+
+    @FindBy(xpath = "//*[@role='dialog']")
+    WebElement dialog;
+
+    @FindBy(xpath = "//*[@class='dropdown-menu show']")
+    WebElement menuDropdown;
+
+    @FindBy(xpath = "//*[@ng-reflect-ngb-tooltip='Permanently deleting']")
+    WebElement deleteButton;
+
+    public void waitForLoading() {
+        try {
+            getWait().forInvisibility(emailDialog);
+            getWait().forVisibility(emailDialog);
+        } catch (StaleElementReferenceException e) {
+            e.printStackTrace();
+        }
+        getWait().forClickable(saveButton);
+    }
+
+    public void openAddDialog() {
+        addEmailButton.click();
+    }
+
+    public void setEmail(String email) {
+        emailDialog.click();
+        emailDialog.clear();
+        emailDialog.sendKeys(email);
+    }
+
+    public void clickSaveButton() {
+        saveButton.click();
+    }
+
+    public boolean makeCellEmail(String email) {
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+
+        try {
+            WebElement element = wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[contains(@ng-reflect-result, '" + email + "')]")));
+            return element.isDisplayed();
+        } catch (NoSuchElementException e) {
+            return false;
+        }
+    }
+
+
+    public void openDropdown(String email) {
+        driver.findElement(By.xpath("//*[contains(@ng-reflect-result, '" + email + "')]/ancestor::tr//*[@class='nav-item w-5 dropdown']")).click();
+    }
+
+    public void clickOnSortLink(SortValues sortValue) {
```

Comment:
SortValues sortValue
This parameter doesn't need, because it unused
And you can delete SortValues enum , you doesn't use value in test

Reply:
Ok, I resolve this enum and parameter, its really doesn't used and need, and after that test doesn't broken
thanks